packages/babel-core/src/transformation/normalize-file.js

```diff
@@ -83,6 +83,27 @@ export default function* normalizeFile(
     }
   }
 
+  if (ast) {
+    if (ast.type === "Program") {
+      ast = t.file(ast, [], []);
+    } else if (ast.type !== "File") {
+      throw new Error("AST root must be a Program or File node");
+    }
+
+    const { cloneInputAst } = options;
+    if (typeof cloneInputAst !== "boolean") {
+      throw Error("cloneInputAst should be a boolean");
```

Comment:
Haven't `cloneInputAst` been validated in `packages/babel-core/src/config/validation/options.js`?

Reply:
Removed additional check :).